File: backend/app/schemas/stats.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, computed_field

from app.core.admin_labels_zh import label_admin_action, label_resource_type


class StatsSummaryOut(BaseModel):
    sites_total: int
    sites_by_category: dict[str, int]
    sites_valid: int
    sites_invalid: int
    articles_total: int
    articles_by_category: dict[str, int]
    articles_by_content_type: dict[str, int]
    visits_total: int
    visits_today: int
    site_clicks_total: int
    article_reads_total: int


class LogActionOption(BaseModel):
    value: str
    label: str


class StatsLogActionsOut(BaseModel):
    items: list[LogActionOption]


class TrendPoint(BaseModel):
    label: str
    count: int


class TrendPageOut(BaseModel):
    items: list[TrendPoint]
    total: int
    page: int
    page_size: int


class AdminLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    admin_id: int | None
    admin_username: str | None = None
    action: str
    resource_type: str | None
    resource_id: str | None
    detail: str | None
    ip: str | None
    created_at: datetime

    @computed_field
    @property
    def action_label(self) -> str:
        return label_admin_action(self.action)

    @computed_field
    @property
    def resource_type_label(self) -> str:
        return label_resource_type(self.resource_type)


class AdminLogPageOut(BaseModel):
    items: list[AdminLogOut]
    total: int
    page: int
    page_size: int

File: frontend/src/api/admin/stats.ts
import http from '@/api/http'
import type { PageResult } from '@/types/models'

export type StatsSummary = {
  sites_total: number
  sites_valid: number
  sites_invalid: number
  sites_by_category: Record<string, number>
  articles_total: number
  articles_by_category: Record<string, number>
  articles_by_content_type: Record<string, number>
  visits_total: number
  visits_today: number
  site_clicks_total: number
  article_reads_total: number
}

export type TrendPoint = { label: string; count: number }

export type AdminLogRow = {
  id: number
  admin_id: number | null
  admin_username: string | null
  action: string
  action_label: string
  resource_type: string | null
  resource_type_label: string
  resource_id: string | null
  detail: string | null
  ip: string | null
  created_at: string
}

export type LogActionOption = { value: string; label: string }

export async function fetchStatsSummary() {
  const { data } = await http.get<StatsSummary>('/api/admin/stats/summary')
  return data
}

export async function fetchStatsLogActions() {
  const { data } = await http.get<{ items: LogActionOption[] }>('/api/admin/stats/log-actions')
  return data.items
}

export type TrendParams = {
  period: 'day' | 'week' | 'month'
  page: number
  page_size: number
}

export async function fetchStatsTrend(params: TrendParams) {
  const { data } = await http.get<PageResult<TrendPoint>>('/api/admin/stats/trend', { params })
  return data
}

export type AdminLogsParams = {
  action?: string
  page: number
  page_size: number
}

export async function fetchAdminLogs(params: AdminLogsParams) {
  const { data } = await http.get<PageResult<AdminLogRow>>('/api/admin/logs', { params })
  return data
}
